Learn how to fix the issue when events are not consumed or are consumed slowly (#1820)
spryker/spryker-docs: docs/scos/dev/troubleshooting/troubleshooting-general-technical-issues/rabbitmq-zed.critical-phpamqplib-exception-amqpchannelclosedexception-channel-connection-is-closed.md

## Description

> Events are not consumed or are consumed much slowler as normal. In RabbitMQ exception.log, the following exception can be found (often also with Broken Pipe reference)

Reviewer: ```suggestion
Events are not consumed or are consumed much slower than normal. In RabbitMQ exception.log, the following exception can be found (often also with Broken Pipe reference):
```

> ```

Reviewer: ```suggestion
```php
```

Zed.CRITICAL: PhpAmqpLib\Exception\AMQPChannelClosedException - Channel connection is closed.
```

## Cause

> There are several potential reasons for this error. The most common cause is that during P&S chunks take too much time to be processed and RabbitMQ is closing its TCP connection as it anticipates either a timeout or no futher connection to happen.

Reviewer: ```suggestion
There are several potential reasons for this error. The most common cause is that during P&S chunks take too much time to be processed and RabbitMQ is closing its TCP connection as it anticipates either a timeout or no further connection to happen.
```

## Solution

It is best to profile the job where you are experiencing this error to understand what exactly makes processing the chunks expensive. Temporary mitigation of the issue might be possible as explained below:

**Adjusting CHUNK_SIZE**

> Using smaller chunk sizes might help to alleviate the issue as it reduces the time until a chunk is fully processed. If you are running a standard publishing setup you can adjust the following value in config_default.php. 

Reviewer: ```suggestion

Using smaller chunk sizes might help to alleviate the issue because it reduces the time until a chunk is fully processed. If you are running a standard publishing setup, you can adjust the following value in `config_default.php`. 
```

> ```

Reviewer: ```suggestion
```php
```
